tutorials/btp-app-events-app-setup-s4hc/btp-app-events-app-setup-s4hc.md

```diff
@@ -0,0 +1,378 @@
+            await UPDATE(Risks).set({status_value: 'ASSESSED'}).where({ID: riskData.ID});
+        });
+
+        bupa.on( 'Created', async (msg) => {
+            const { BusinessPartner } = msg.data;
+            LOG.info('Received created! BusinessPartner=' + BusinessPartner);
+            await createRisk(BusinessPartner);
+        });
+
+        bupa.on( 'Changed', async (msg) => {
+            const { BusinessPartner } = msg.data;
+            LOG.info('Received changed! BusinessPartner=' + BusinessPartner);
+            if((await SELECT.one.from(Risks).where({supplier_ID: BusinessPartner})).status_value === 'NEW') return;
+            await UPDATE(Risks).set({status_value: 'CHANGED'}).where({'supplier_ID' : BusinessPartner});
+        });
+
+        async function createRisk(businessPartner) {
+            const payload = {
+                title: 'auto: CFR non-compliance',
+                descr: 'New Business Partner might violate CFR code',
+                prio: '1',
+                impact: 200000,
+                supplier_ID: businessPartner,
+                status_value: 'NEW'
+            }
+            LOG.info("Creating auto risk with", payload);
+            await INSERT.into(Risks).entries(payload);
+        }
+    });
+    ```
+
+6. To add the new status column to the table in the work list page and the object page, open the `srv/risks-service-ui.cds` file and add the following lines:
+
+    ```JavaScript[9,25-28,36]
+    using RiskService from './risk-service';
+
+    annotate RiskService.Risks with {
+      title       @title: 'Title';
+      prio        @title: 'Priority';
+      descr       @title: 'Description';
+      miti        @title: 'Mitigation';
+      impact      @title: 'Impact';
+      status      @readonly @title: 'Status';
+      supplier    @(
+            title: 'Supplier',
+            Common.Text: supplier.fullName,
+            Common.TextArrangement: #TextOnly
+        )
+    }
+    ...
+    annotate RiskService.Risks with @(
+      UI: {
+        ...
+        LineItem: [
+          ...
+          {
+            Value: impact,
+            Criticality: criticality
+          },
+          {
+            Value:          status_value,
+            Criticality: status.criticality
+          }
+        ]
+        ...
+        FieldGroup#Main: {
+          Data: [
+            ...
+            {Value: supplier.isBlocked},
+            {Value: status_value, Criticality: status.criticality},
+          ]
+    ...
+    ```
+
+[DONE]
+[ACCORDION-END]
+---
+[ACCORDION-BEGIN [Step 3: ](Test your changes locally)]
+So, we have added logic to create a new risk when an event for a newly created supplier is received. We have also added logic to update our risks when an event for an updated supplier is received from our SAP HANA Cloud system. This is achieved by listening to events from the SAP S/4HANA Cloud system.
+
+Let's simulate locally business partner creation and update, and see the results:
```

Comment:
"Let's" should be removed, or replaced with "Now"

Reply:
done